src/wabt.cpp

```diff
@@ -103,125 +103,1084 @@ ExecutionResult WabtEngine::execute(
+      const interp::TypedValues& args,
+      interp::TypedValues& results
+    ) {
+      uint64_t number = static_cast<uint64_t>(args[0].value.i64);
+      uint32_t resultOffset = static_cast<uint32_t>(args[1].value.i32);
+      results[0].set_i32(static_cast<uint32_t>(interface.eeiGetBlockHash(number, resultOffset)));
+      return interp::Result::Ok;
+    }
+  );
+
+  hostModule->AppendFuncExport(
+    "call",
+    {{Type::I64, Type::I32, Type::I32, Type::I32, Type::I32}, {Type::I32}},
+    [&interface](
+      const interp::HostFunc*,
+      const interp::FuncSignature*,
+      const interp::TypedValues& args,
+      interp::TypedValues& results
+    ) {
+      int64_t gas = static_cast<int64_t>(args[0].value.i64);
+      uint32_t addressOffset = static_cast<uint32_t>(args[1].value.i32);
+      uint32_t valueOffset = static_cast<uint32_t>(args[2].value.i32);
+      uint32_t dataOffset = static_cast<uint32_t>(args[3].value.i32);
+      uint32_t dataLength = static_cast<uint32_t>(args[4].value.i32);
+      results[0].set_i32(static_cast<uint32_t>(interface.eeiCall(EthereumInterface::EEICallKind::Call, gas, addressOffset, valueOffset, dataOffset, dataLength)));
+      return interp::Result::Ok;
+    }
+  );
+
+  hostModule->AppendFuncExport(
+    "callDataCopy",
+    {{Type::I32, Type::I32, Type::I32}, {}},
+    [&interface](
+      const interp::HostFunc*,
+      const interp::FuncSignature*,
+      const interp::TypedValues& args,
+      interp::TypedValues&
+    ) {
+      uint32_t resultOffset = static_cast<uint32_t>(args[0].value.i32);
+      uint32_t dataOffset = static_cast<uint32_t>(args[1].value.i32);
+      uint32_t length = static_cast<uint32_t>(args[2].value.i32);
+      interface.eeiCallDataCopy(resultOffset, dataOffset, length);
+      return interp::Result::Ok;
+    }
+  );
+
+  hostModule->AppendFuncExport(
+    "getCallDataSize",
+    {{}, {Type::I32}},
+    [&interface](
+      const interp::HostFunc*,
+      const interp::FuncSignature*,
+      const interp::TypedValues&,
+      interp::TypedValues& results
+    ) {
+      results[0].set_i32(interface.eeiGetCallDataSize());
+      return interp::Result::Ok;
+    }
+  );
+
+  hostModule->AppendFuncExport(
+    "callCode",
+    {{Type::I64, Type::I32, Type::I32, Type::I32, Type::I32}, {Type::I32}},
+    [&interface](
+      const interp::HostFunc*,
+      const interp::FuncSignature*,
+      const interp::TypedValues& args,
+      interp::TypedValues& results
+    ) {
+      int64_t gas = static_cast<int64_t>(args[0].value.i64);
+      uint32_t addressOffset = static_cast<uint32_t>(args[1].value.i32);
+      uint32_t valueOffset = static_cast<uint32_t>(args[2].value.i32);
+      uint32_t dataOffset = static_cast<uint32_t>(args[3].value.i32);
+      uint32_t dataLength = static_cast<uint32_t>(args[4].value.i32);
+      results[0].set_i32(static_cast<uint32_t>(interface.eeiCall(EthereumInterface::EEICallKind::CallCode, gas, addressOffset, valueOffset, dataOffset, dataLength)));
```

Comment:
There are a lot of unnecessary casts here (in the whole file).

1. The `args[].value.i32` is `uint32_t` so you don't have to cast it explicitly.
2. You can also use `.get_i32()` instead of `.value.i32`. Not shorted, but similar to `.set_i32()`.
3. The `eeiCall()` returns `uint32_t`, no cast needed. Similarly for other functions as already reported.